### PR TITLE
LLE-10148 LLE-10149: Add document edges table, edge computation, and graph query functions

### PR DESCRIPTION
**Pull request by GSasu**
## Summary

- Add `memory.document_edges` table migration for pre-computed document relationships (9 edge types, RLS, FK cascade, indexes)
- Add edge computation functions to `@jeffs-brain/memory-postgres`: centroid calculation (AVG with vector cast for halfvec), cosine similarity via pgvector, shared tags (Jaccard), shared folders, same session, supersedes, session-episode, episode-heuristic, document-ontology, and wikilink parsing
- Add graph query functions: `getDocumentGraph` (7 entity types, filtered edges, communities placeholder) and `getDocumentStats` (SQL GROUP BY aggregation)
- Bulk upsert via `unnest()` pattern, transaction-wrapped delete+compute+upsert, embedding dimension support (1024/3072), scoped edge deletion

## Test plan

- [ ] 12 integration tests pass (75 assertions)
- [ ] `bunx tsc -p sdks/ts/memory-postgres/tsconfig.json --noEmit` passes
- [ ] `bun run build` passes
- [ ] Edge cases covered: no chunks, no embeddings, no tags, empty brain, all 7 entity types, all 9 edge types
- [ ] Centroid uses `avg(embedding::vector)` cast (not raw `avg(halfvec)`)
- [ ] Both 1024 and 3072 embedding dimensions handled
- [ ] No `any` types, no American English, no SQL injection risk

## Summary by CodeRabbit

* **New Features**
  * Persistent, weighted document relationships across many relation types (semantic similarity, shared tags/folders, same-session, supersedes, session↔episode, episode heuristics, ontology links, wikilinks).
  * Document graph API: node/edge retrieval, filtering by edge/entity types, counts, truncation indicators, and last-activity timestamps.
  * Public utilities to compute, bulk upsert, find, and delete document edges.

* **Chores**
  * DB migration: new tenant-isolated document_edges table with indexes and row-level security.
  * Unified LLM request shape for system/user prompts to standardize streaming and extraction behavior.

* **Tests**
  * Extensive integration/unit tests covering edge computation, persistence, graph queries, filtering, and stats.

[Review Change Stack](https://app.coderabbit.ai/change-stack/jeffs-brain/memory/pull/73?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: go/graph/edges.go [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+	sharedFolderWeight     = 0.3
+	sameSessionWeight      = 0.5
+	supersedesWeight       = 1.0
+	sessionEpisodeWeight   = 0.8
+	episodeHeuristicWeight = 0.7
+	documentOntologyWeight = 0.9
+	wikilinkWeight         = 0.7
+)
+
+// SimilarityOptions controls nearest-neighbor semantic edge computation.
```
[the remark this answers is not in the document]
Comment (GSasu) [reply go/graph/edges.go]: Fixed in fd1a46f. Updated comment wording to fix the misspelling that was blocking golangci-lint.

File: go/graph/edges.go [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+			where trim(split_part(matches[1], '|', 1)) <> ''
+		)
+		select d.document_id::text, $4::double precision, l.target
+		from memory.documents d
+		join links l
+		  on d.path = l.target
+		  or d.path = l.target || '.md'
+		  or d.path like '%/' || l.target
+		  or d.path like '%/' || l.target || '.md'
+		where d.brain_id = $2::uuid and d.tenant_id = $3::uuid and d.document_id != $1::uuid`,
```
[the remark this answers is not in the document]
Comment (GSasu) [reply go/graph/edges.go]: Fixed in 4998391. LIKE wildcard characters (% and _) are now escaped in wikilink path matching, consistent with the supersedes logic.

File: go/graph/graph.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+		 and c.tenant_id = d.tenant_id
+		where d.brain_id = $1::uuid
+		  and d.tenant_id = $2::uuid
+		group by d.document_id, d.path, d.size, d.updated_at, d.metadata
+		order by d.updated_at desc
+		limit $3`,
+		brainID,
+		tenantID,
+		maxNodes,
+	)
```
[the remark this answers is not in the document]
Comment (GSasu) [reply go/graph/graph.go:81]: Fixed in fd1a46f. Truncation detection now fetches limit+1, trims to limit, and sets Truncated only when the extra row exists.
